src/datasets/arrow_dataset.py

```diff
@@ -2651,7 +2652,19 @@ def validate_function_output(processed_inputs, indices):
                     f"Provided `function` which is applied to all elements of table returns a variable of type {type(processed_inputs)}. Make sure provided `function` returns a variable of type `dict` (or a pyarrow table) to update the dataset or `None` if you are only interested in side effects."
                 )
             elif isinstance(indices, list) and isinstance(processed_inputs, Mapping):
-                allowed_batch_return_types = (list, np.ndarray)
+                allowed_batch_return_types = (list, np.ndarray, pd.Series)
+                if config.TF_AVAILABLE and "tensorflow" in sys.modules:
```

Comment:
Maybe a silly question: is it needed to check that the module has been previously loaded `in sys.modules`?

Reply:
It's more a quality of life. The check is not needed if the module has not been imported.

And this way the user is not suprised to see TF imported even though they're not using it. Especially since importing TF takes several seconds